Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -191,19 +193,27 @@ def validate_enforcement_boundary_create_arguments(
         raise InvalidValueError("protocol", protocol, VALID_PROTOCOLS)
 
 
-def validate_ip_lists_get_arguments(max_results: Optional[int]) -> None:
+def validate_ip_lists_get_arguments(max_results: Optional[int], ip_address: Optional[str]) -> None:
     """Validate arguments for ip-lists-get command.
 
     Args:
         max_results: Number of maximum results returned.
+        ip_address: IP address of ip lists to be returned.
     """
+    IP_REGEX = "^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$"
```

Comment:
This regex approach is overcomplicated - you can use the `ipaddress` builtin module to verify both IPv4 and IPv6 addresses, e.g.:

```
import ipaddress
try:
    ipaddress.ip_address(ip_address)
except:
    ...
```

Reply:
We also do have an inbuilt function in commonserver python, i.e. `is_ip_valid`, we can use that as well.

Reply:
@yashgaikwad-crest - at a minimum, we should be passing `accept_v6_ips=true` to the `is_ip_valid` builtin: https://xsoar.pan.dev/docs/reference/api/common-server-python#is_ip_valid

The implementation is pretty restrictive here though, checking explicitly for 4 octets if passed an IPv4 address: https://github.com/demisto/content/blob/6031bbc9f658df84fa8daecf9ea1f8c1b8524e8a/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L2707

The ip_lists GET accepts partial matches/octets (which we don't necessarily need to support, but better to explicitly decide to only support full addresses than to have this incidental limitation) which will fail this check. For this, `socket.inet_aton` is better for IPv4 validation, so the optimal solution is probably to leverage the XSOAR builtin `is_ipv6_valid` for the `socket.inet_pton` check and then check with `socket.inet_aton` on failure:

```
if not is_ipv6_valid(ip_address):
    try:
        socket.inet_aton(ip_address)
    except:
        raise ...
```

Reply:
Yes, we've made the changes accordingly, first will validate to IP & on failure will check for `socket.inet_aton`